stellar/ttl.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+const { Contract, SorobanRpc, SorobanDataBuilder, TransactionBuilder, Operation } = require('@stellar/stellar-sdk');
+const { Server } = require('@stellar/stellar-sdk/rpc');
+const { Command } = require('commander');
+const { execSync } = require('child_process');
+const { loadConfig, printInfo, saveConfig } = require('../evm/utils');
+const { stellarCmd, getNetworkPassphrase, getWallet, addBaseOptions } = require('./utils');
+const { getChainConfig, addOptionsToCommands } = require('../common');
+const { prompt } = require('../common/utils');
+require('./cli-utils');
+
+const MAX_INSTANCE_TTL_EXTENSION = 535679;
+
+function checkContractName(chain, contractName) {
+    if (!chain.contracts?.[contractName]) {
+        throw new Error('Contract not found.');
+    }
+}
+
+async function getTtl(chain, contractName, _args, _options) {
+    printInfo('get ttl', contractName);
+    const ledgerEntry = await getLedgerEntry(chain, contractName);
+    printInfo('latest ledger', ledgerEntry.latestLedger);
+    printInfo('live until ledger', ledgerEntry.entries[0].liveUntilLedgerSeq);
+}
+
+async function getLedgerEntry(chain, contractName) {
+    const contract = new Contract(chain.contracts[contractName].address);
+    const instance = contract.getFootprint();
+    const server = new SorobanRpc.Server(chain.rpc);
+    return await server.getLedgerEntries(...[instance]);
+}
+
+async function extendInstance(chain, contractName, _args, options) {
+    checkContractName(chain, contractName);
+
+    const { yes } = options;
+    const { rpc, networkType } = chain;
+
+    if (prompt(`Extend instance ttl for ${contractName}`, yes)) {
+        return;
+    }
+
+    const server = new Server(rpc);
+    const wallet = await getWallet(chain, options);
+
+    const account = await server.getAccount(wallet.publicKey());
+    const fee = '200100'; // Base fee plus resource fee
+
+    const contract = new Contract(chain.contracts[contractName].address);
+    const instance = contract.getFootprint();
+
+    const sorobanData = new SorobanDataBuilder().setResourceFee(200000).setReadOnly([instance]).build();
+    const transaction = new TransactionBuilder(account, {
```

Comment:
There is `buildTransaction` and `prepareTransaction` function, It would be better to utilize it

Reply:
I'm a little concerned about the existing `buildTransaction` function, as it doesn't provide the flexibility to ` .setReadOnly([instance])`, which is necessary for the `ExtendFootprintTTLOp`. What do you think we should do here?

Reply:
yeah, I think you can pass `setSorobanData` as optional parameter, update `buildTransaction` something like:
```
async function buildTransaction(operation, server, wallet, networkType, options = {}) {
    const account = await server.getAccount(wallet.publicKey());
    const networkPassphrase = getNetworkPassphrase(networkType);
    const transactionBuilder = new TransactionBuilder(account, {
        fee: BASE_FEE,
        networkPassphrase,
    })
        .addOperation(operation)
        .setTimeout(options.timeout || 30);
    
    if (options.sorobanData) {
        transactionBuilder.setSorobanData(options.sorobanData);
    }

    const builtTransaction = transactionBuilder.build();

    if (options.verbose) {
        printInfo('Tx', builtTransaction.toXDR());
    }

    return builtTransaction;
}
```

Reply:
As per discussion in SC meeting, changing command to run CLI contract extend command rather than alter existing JS SDK helper functions.